.circleci/config.yml

```diff
@@ -62,46 +62,39 @@ jobs:
             cd build
             make install
   build_gpu:
+    parameters:
+      cuda_version:
+        type: string
     machine:
       resource_class: gpu.nvidia.small
       image: ubuntu-1604-cuda-10.1:201909-23
+      docker_layer_caching: true
     steps:
       - checkout
-      - run:
-          name: Install apt packages
-          command: |
-            sudo apt-get update
-            sudo apt-get install -y git-core build-essential cmake
       - run:
           name: Initialize submodules
           command: |
             git submodule init
             git submodule update
       - run:
-          name: Build
-          command: |
-            mkdir build
-            cd build
-            cmake ../ \
-              -DCMAKE_C_FLAGS="-Werror -Wno-deprecated-declarations" \
-              -DCMAKE_CXX_FLAGS="-Werror -Wno-deprecated-declarations" \
-              -DTP_ENABLE_SHM=OFF \
-              -DTP_ENABLE_CMA=OFF \
-              -DTP_USE_CUDA=ON \
-              -DTP_ENABLE_CUDA_IPC=ON \
-              -DTP_ENABLE_IBV=OFF \
-              -DTP_BUILD_TESTING=ON
-            make -j20
-      - run:
-          name: Test
+          name: Build/test
           command: |
-            cd build
-            ./tensorpipe/test/tensorpipe_test --gtest_filter=-*CudaGdr*
-      - run:
-          name: Install
-          command: |
-            cd build
-            sudo make install
+              docker build -t tensorpipe -f .circleci/Dockerfile.cuda<< parameters.cuda_version >> .
+              docker run --gpus all tensorpipe sh -c "
+                mkdir build && cd build &&
+                cmake ../ \
+                  -DCMAKE_C_FLAGS=\"-Werror -Wno-deprecated-declarations\" \
+                  -DCMAKE_CXX_FLAGS=\"-Werror -Wno-deprecated-declarations\" \
+                  -DCUDA_NVCC_FLAGS=\"-gencode arch=compute_61,code=sm_61\" \
```

Comment:
This line is new, why is it needed?

Reply:
It is not needed strictly speaking, but it allows for generating device code optimized for the GPU type available in CircleCI instances.